### ISSUE TITLE
IE11 Compatibility

### ISSUE THREAD

**Issue by shaunanoordin** [zooniverse/anti-slavery-manuscripts#18]
## Issue Overview
* We currently have not implemented compatibility with Internet Explorer 11.
* There are two parts to consider:
  * Polyfills (e.g. [Polyfill.io](https://polyfill.io/v2/docs/), notably for Promises) should take care of the _functionality_ side of things.
  * As for the presentation side of things, the CSS may require more hacky solutions - but we'll cross those bridges and/or set fire to them when necessary.
* More important question: _do we want IE11 compatibility?_ We already have Edge compatibility (woohoo!) and I'm not sure what the user base for this project will be like, and I generally prefer adding IE11 hacks unless absolutely necessary.

**Comment by srallen** [comment]
Our policy has always been supporting current version and the most recent previous version of the major browser vendors, which should mean Edge 14 and 15 and IE 11 is deprecated. However, adoption of Edge hasn't been widespread yet enough. I think adding IE 11 functional fixes using polyfills is inconsequential and thus should be done, but I wouldn't spend too much time on the CSS side of things. Ideally, using something like Grommet would handle this for us too.

**Comment by shaunanoordin** [comment]
Update: I brought up the direction to ditch IE11 for Anti-Slavery Manuscripts (aka Liberating the Liberator) during today's Zoominar, but unfortunately the answer is _IE11 support is necessary for the targeted user base of this project._

Moving forward, we'll need to sort out the IE11 compatibility issues... and my official reaction is [this](https://giphy.com/gifs/sad-depressed-disappointed-33iqmp5ATXT5m]

**Comment by shaunanoordin** [comment]
Note: as @wgranger pointed out in PR #29, there was an IE11 workaround used for the Contrast functionality. Once we get IE11 compatibility running, and if we still have issues with the Contrast Subject Image button, the solution (in SVG image) was to dynamically insert `<svg><defs><filter>` into the document body.

```
const INVERT =
  `<svg style="position: fixed; right: 100%; top: 100%; visibility: hidden;">
    <defs>
      <filter id="svg-invert-filter">
        <feComponentTransfer>
          <feFuncR type="table" tableValues="1 0"/>
          <feFuncG type="table" tableValues="1 0"/>
          <feFuncB type="table" tableValues="1 0"/>
        </feComponentTransfer>
      </filter>
    </defs>
  </svg>`;

invertFilter() {
  if (this.props.contrast) {
    if (!document.getElementById('svg-invert-filter')) {
      document.body.insertAdjacentHTML('afterbegin', INVERT);
    }
    ...
  }
  ...
}
```

**Comment by shaunanoordin** [comment]
Checkpoint: as of Commit [5769837](https://github.com/zooniverse/liberating-the-liberator/commit/5769837063353f32b9a43de362517a8089859a65), most functionality and some issues have been confirmed for _Win7+IE11 and Win10+Edge on localhost._

TL;DR: IE11/Edge Compatibility looks good for now.

Details:
* User can use the Classify page.
  * User can navigate (zoom, translate, rotate, reset) the Subject image.
  * User can make new Annotations.
    * Issue: When examining existing Annotations, the text appears blank. This is consistent across all browsers so isn't a compatibility issue,
      * Hypothesis: the text data for each Annotation isn't saved; presumably this is addressed in the upcoming PR #61 
    * Feature incomplete: User cannot submit Classifications. See PR #61.
  * User can access the Toggle Previous Marks, Contrast, and Subject Info features.
    * Minor IE11 compatibility issue: the Subject Info window is anchored to the top-left corner of the page an is immovable. Otherwise, it's functional enough.
* User can navigate between pages using the main navi.
* Home page looks OK, although missing some content.
* About page looks OK, although missing some content.
